File: create_test_data.py
import xlsxwriter
from datetime import datetime, timedelta
import random

# 创建测试数据
workbook = xlsxwriter.Workbook('test_stock_data.xlsx')
worksheet = workbook.add_worksheet()

# 写入表头
worksheet.write(0, 0, '日期')
worksheet.write(0, 1, '收盘价')

# 生成100天的测试数据
start_date = datetime(2024, 1, 1)
base_price = 100

for i in range(100):
    date = start_date + timedelta(days=i)
    # 随机价格波动
    price = base_price + random.uniform(-10, 10)
    base_price = price

    worksheet.write(i + 1, 0, date.strftime('%Y-%m-%d'))
    worksheet.write(i + 1, 1, round(price, 2))

workbook.close()
print("测试文件已创建: test_stock_data.xlsx")

File: create_test_data.js
const XLSX = require('xlsx');
const fs = require('fs');

// 生成测试数据
const data = [];
data.push(['日期', '收盘价']);

let basePrice = 100;
const startDate = new Date(2024, 0, 1); // 2024-01-01

for (let i = 0; i < 100; i++) {
    const date = new Date(startDate);
    date.setDate(startDate.getDate() + i);

    const year = date.getFullYear();
    const month = String(date.getMonth() + 1).padStart(2, '0');
    const day = String(date.getDate()).padStart(2, '0');
    const dateStr = `${year}-${month}-${day}`;

    // 随机价格波动
    const price = basePrice + (Math.random() - 0.5) * 20;
    basePrice = price;

    data.push([dateStr, parseFloat(price.toFixed(2))]);
}

// 创建工作簿
const ws = XLSX.utils.aoa_to_sheet(data);
const wb = XLSX.utils.book_new();
XLSX.utils.book_append_sheet(wb, ws, 'Sheet1');

// 保存文件
XLSX.writeFile(wb, 'test_stock_data.xlsx');
console.log('测试文件已创建: test_stock_data.xlsx');
